libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1418,6 +1418,10 @@ bool TypeChecker::visit(TupleExpression const& _tuple)
 				components[i]->accept(*this);
 				types.push_back(type(*components[i]));
 
+				if (types[i]->category() == Type::Category::Tuple)
+					if (dynamic_cast<TupleType const&>(*types[i]).components().empty())
+						m_errorReporter.fatalTypeError(components[i]->location(), "Type of tuple component cannot be null.");
```

Comment:
I know I used the world `null` in the description of the issue, but this is actually the first place this appears. Perhaps we should say something like "non-empty value"?

Reply:
I.e. "cannot be empty"?

Reply:
Also, I think it should only be a warning pre-0.5.0

Reply:
I'd be up for "cannot be empty". Do you suggest this to be a warning rather than an error? I think we should treat this as an error.

Reply:
Let's again make it a warning pre-050 and an error starting from 050